### ISSUE TITLE
GUI: Support for texture_swizzle

### ISSUE THREAD

**Issue by JayFoxRox** [apitrace/apitrace#373]
It would be nice if the GUI could respect the texture swizzle masks (for color channels) in the image preview and image viewer [possibly make it an option?].

**Comment by jrfonseca** [comment]
What "texture swizzles" are you referring to?

**Comment by JayFoxRox** [comment]
https://www.opengl.org/registry/specs/EXT/texture_swizzle.txt
https://www.opengl.org/registry/specs/ARB/texture_swizzle.txt
https://www.opengl.org/wiki/Texture#Swizzle_mask
D3D probably has a counterpart, too?
Sorry for not being more specific earlier.

**Comment by jrfonseca** [comment]
I see.  So IIUC, you want the textures seen on the GUI to respect `TEXTURE_SWIZZLE_RGBA` sampler state.

I see that how that could be useful sometimes, but it could also be equally confusing -- imagine the bug is that `TEXTURE_SWIZZLE_RGBA` was not set properly: if the GUI swizzles will give the illusion that the texture data itself may be wrong.

The `TEXTURE_SWIZZLE_RGBA` state, and a lot of other sampler state is visible in the parameters tab.

Given I'm not too sure about this feature, and that I already have more bugs/features that I'm sure about  than time to fix them, I'm afraid that in practice this is a feature I'll never do.

If more people chip in and request this feature then I'll reconsider.  But in the meanwhile there is no point in keeping this open if no further action is planned.

**Comment by JayFoxRox** [comment]
I'll probably work on this myself then [and make it an option in the GUI].
If the state is set up wrong the texture fetches will also be bad.
On the other hand: If the texture swizzle is not implemented in apitrace our xbox alpha textures will be red / black (GL_R8) instead of alpha (using a ONE,ONE,ONE,RED swizzle) in the surfaces tab / image viewer.
I'd expect to see the texture how the shaders receive them [= swizzle option turned on] + how they were uploaded [= swizzle option off].

Thanks for looking into it anyway :)

**Comment by jrfonseca** [comment]
Yes, you could have glretrace to dump the swizzles next to the raw image, somewhere in the JSON object, so the GUI can toggle them without asking glretrace to dump textures again.

BTW, the place to do this would be in `glretrace_images.cpp`'s `dumpActiveTextureLevel` function.
